packages/core/parcel-bundler/src/assets/CSSAsset.js

```diff
@@ -125,6 +148,48 @@ class CSSAsset extends Asset {
         'module.exports = ' + JSON.stringify(this.cssModules, null, 2) + ';';
     }
 
+    let map;
+    if (this.options.sourceMaps) {
+      if (this.sourceMap) {
+        if (this.sourceMap instanceof SourceMap) {
+          map = this.sourceMap;
+        } else {
+          map =
+            typeof this.sourceMap === 'string'
+              ? JSON.parse(this.sourceMap)
+              : this.sourceMap;
+
+          map = await new SourceMap().addMap(map);
+
+          if (this.sourceMap.toJSON) {
+            // this.sourceMap instanceof SourceMapGenerator
+            let sourceLines = this.contents.split('\n');
+            map.mappings = map.mappings.filter(
```

Comment:
PostCSS seems to generate sourcemaps with invalid source positions: [with this visualizer](http://sokra.github.io/source-map-visualization/#base64,ZGl2e3dpZHRoOjEwMHB4O2hlaWdodDoxMDBweH1tYWlue2ZvbnQtZmFtaWx5Om1vbm9zcGFjZX1ib2R5e2JhY2tncm91bmQtY29sb3I6cmVkfQovKiMgc291cmNlTWFwcGluZ1VSTD0vc3R5bGUuY3NzLm1hcCov,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIm90aGVyLXN0eWxlLmNzcyIsImFub3RoZXItc3R5bGUuY3NzIiwic3R5bGUuY3NzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUNBLElBQ0MsWUFBYSxBQUNiLFlBQWMsQ0FDZCxBQ0pELEtBQ0MscUJBQXVCLENBQ3ZCLEFDQ0QsS0FDQyxvQkFBc0IsQ0FDdEIiLCJmaWxlIjoic3R5bGUuY3NzLm1hcCIsInNvdXJjZVJvb3QiOiIuLi9pbnRlZ3JhdGlvbi9zb3VyY2VtYXAtY3NzLWltcG9ydCIsInNvdXJjZXNDb250ZW50IjpbIlxuZGl2IHtcblx0d2lkdGg6IDEwMHB4O1xuXHRoZWlnaHQ6IDEwMHB4O1xufVxuIiwibWFpbiB7XG5cdGZvbnQtZmFtaWx5OiBtb25vc3BhY2U7XG59XG4iLCJcbkBpbXBvcnQgXCIuL290aGVyLXN0eWxlLmNzc1wiO1xuXG5ib2R5IHtcblx0YmFja2dyb3VuZC1jb2xvcjogcmVkO1xufVxuXG5AaW1wb3J0IFwiLi9hbm90aGVyLXN0eWxlLmNzc1wiO1xuIl19,CmRpdiB7Cgl3aWR0aDogMTAwcHg7CgloZWlnaHQ6IDEwMHB4Owp9Cg==,bWFpbiB7Cglmb250LWZhbWlseTogbW9ub3NwYWNlOwp9Cg==,CkBpbXBvcnQgIi4vb3RoZXItc3R5bGUuY3NzIjsKCmJvZHkgewoJYmFja2dyb3VuZC1jb2xvcjogcmVkOwp9CgpAaW1wb3J0ICIuL2Fub3RoZXItc3R5bGUuY3NzIjsK), these mappings are clearly visible (by hovering over the highlighted entries in the bottom section).
With the filtering, [this](http://sokra.github.io/source-map-visualization/#base64,ZGl2e3dpZHRoOjEwMHB4O2hlaWdodDoxMDBweH1tYWlue2ZvbnQtZmFtaWx5Om1vbm9zcGFjZX1ib2R5e2JhY2tncm91bmQtY29sb3I6cmVkfQovKiMgc291cmNlTWFwcGluZ1VSTD0vc3R5bGUuY3NzLm1hcCov,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIm90aGVyLXN0eWxlLmNzcyIsImFub3RoZXItc3R5bGUuY3NzIiwic3R5bGUuY3NzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUNBLElBQ0MsWUFDQSxhQ0hELEtBQ0Msc0JDRUQsS0FDQyIsImZpbGUiOiJzdHlsZS5jc3MubWFwIiwic291cmNlUm9vdCI6Ii4uL2ludGVncmF0aW9uL3NvdXJjZW1hcC1jc3MtaW1wb3J0Iiwic291cmNlc0NvbnRlbnQiOlsiXG5kaXYge1xuXHR3aWR0aDogMTAwcHg7XG5cdGhlaWdodDogMTAwcHg7XG59XG4iLCJtYWluIHtcblx0Zm9udC1mYW1pbHk6IG1vbm9zcGFjZTtcbn1cbiIsIlxuQGltcG9ydCBcIi4vb3RoZXItc3R5bGUuY3NzXCI7XG5cbmJvZHkge1xuXHRiYWNrZ3JvdW5kLWNvbG9yOiByZWQ7XG59XG5cbkBpbXBvcnQgXCIuL2Fub3RoZXItc3R5bGUuY3NzXCI7XG4iXX0=,CmRpdiB7Cgl3aWR0aDogMTAwcHg7CgloZWlnaHQ6IDEwMHB4Owp9Cg==,bWFpbiB7Cglmb250LWZhbWlseTogbW9ub3NwYWNlOwp9Cg==,CkBpbXBvcnQgIi4vb3RoZXItc3R5bGUuY3NzIjsKCmJvZHkgewoJYmFja2dyb3VuZC1jb2xvcjogcmVkOwp9CgpAaW1wb3J0ICIuL2Fub3RoZXItc3R5bGUuY3NzIjsK) is the output.

Chrome works with both cases, but the `source-map` package used in the tests doesn't behave as expected with the duplicate & incorrect mapppings.

Reply:
Shouldn't these mappings be updated before converting them to a `SourceMap` instance.

Reply:
`this.sourceMap` might be a SourceMapGenerator, which doesn't expose the mappings. The SourceMap class is "private", so I can just use the mappings there.

Reply:
When? What creates a SourceMapGenerator? `private` doesn't exist in Javascript so not sure what you mean by that.

Reply:
Owww damn thought the fix for null mappings was already in master, anyways this PR should fix this: #2149 afaik

Reply:
Looking at the example you put it the original comment it looks like null mappings. I might be wrong

Reply:
> If that PR doesn't fix it, I'm not sure why it's in those sourcemaps, but it looks to me like it should be fixed on SourceMap level instead of filtering this out. As it's probably a valid sourcemap.

The thing is: the sourcemaps work just fine in Chrome. Testing them is the issue. I could only do the filtering in the tests and leave the invalid mappings in the sourcemap?

Reply:
Yeah that's probably a better idea

Reply:
Please take a look at [this](http://sokra.github.io/source-map-visualization/#base64,Ym9keXtmb250OjEwMCUgSGVsdmV0aWNhLHNhbnMtc2VyaWY7Y29sb3I6IzMzM31ib2R5IGRpdntiYWNrZ3JvdW5kLWNvbG9yOnJlZDt3aWR0aDoxMDBweDtoZWlnaHQ6MTAwcHh9bWFpbntkaXNwbGF5Om5vbmV9Ci8qIyBzb3VyY2VNYXBwaW5nVVJMPS9zdHlsZS5jc3MubWFwICov,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImxpYnJhcnkuc2NzcyIsInN0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFHQSxLQUNFLCtCQUNBLFVBQUEsQ0FFQSxTQUNDLHFCQUNBLFlBQ0EsWUFBQSxDQUFBLEFDUkgsS0FDQyxZQUFjLENBQ2QiLCJmaWxlIjoic3R5bGUuY3NzLm1hcCIsInNvdXJjZVJvb3QiOiIuLi9pbnRlZ3JhdGlvbi9zb3VyY2VtYXAtY3NzLWV4aXN0aW5nIiwic291cmNlc0NvbnRlbnQiOlsiJGZvbnQtc3RhY2s6ICAgIEhlbHZldGljYSwgc2Fucy1zZXJpZjtcbiRwcmltYXJ5LWNvbG9yOiAjMzMzO1xuXG5ib2R5IHtcbiAgZm9udDogMTAwJSAkZm9udC1zdGFjaztcbiAgY29sb3I6ICRwcmltYXJ5LWNvbG9yO1xuXG4gIGRpdiB7XG4gIFx0YmFja2dyb3VuZC1jb2xvcjogcmVkO1xuICBcdHdpZHRoOiAxMDBweDtcbiAgXHRoZWlnaHQ6IDEwMHB4O1xuICB9XG59IiwiQGltcG9ydCBcIi4vbGlicmFyeS5jc3NcIjtcblxubWFpbiB7XG5cdGRpc3BsYXk6IG5vbmU7XG59XG4iXX0=,JGZvbnQtc3RhY2s6ICAgIEhlbHZldGljYSwgc2Fucy1zZXJpZjsKJHByaW1hcnktY29sb3I6ICMzMzM7Cgpib2R5IHsKICBmb250OiAxMDAlICRmb250LXN0YWNrOwogIGNvbG9yOiAkcHJpbWFyeS1jb2xvcjsKCiAgZGl2IHsKICAJYmFja2dyb3VuZC1jb2xvcjogcmVkOwogIAl3aWR0aDogMTAwcHg7CiAgCWhlaWdodDogMTAwcHg7CiAgfQp9,QGltcG9ydCAiLi9saWJyYXJ5LmNzcyI7CgptYWluIHsKCWRpc3BsYXk6IG5vbmU7Cn0K) (from the `should load existing sourcemaps for CSS files` test with minification enabled).

The generated column 102 maps to two valid positions at the same time. 

Filtering it after concatenating doesn't seem to work...

Reply:
Nearly fixed in PostCSS.